server/domain/routers/waterConsumption.go

```diff
@@ -0,0 +1,208 @@
+
+	// router.GET("/waterconsumption/week", WeekWaterConsumption)
+
+	// router.GET("/waterconsumption/month", MonthWaterConsumption)
+
+	// router.GET("/waterconsumption/year", YearWaterConsumption)
+}
+
+// PingWaterConsumption ...
+func PingWaterConsumption(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	waterconsumption := devices.WaterConsumption{}
+	device, err := waterconsumption.Ping()
+	// testing custom error response
+	if err == devices.ErrNotFound {
+		http.Error(w, errors.New("the device is not found").Error(), http.StatusNotFound)
+		return
+	}
+
+	response, err := json.Marshal(device)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusNotFound)
+		return
+	}
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(response)
+
+	//fmt.Fprint(w, fmt.Sprintf("Pong... %v  ---- ERR: %v \n", device, err))
+}
+
+// GetWaterMeterSensor ...
+func GetWaterMeterSensor(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	watermeter := devices.WaterConsumption{}
+	device, err := watermeter.Get()
+	// testing custom error response
+	if err == devices.ErrNotFound {
+		http.Error(w, errors.New("the device is not found").Error(), http.StatusNotFound)
+		return
+	}
+
+	response, err := json.Marshal(device)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusNotFound)
+		return
+	}
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(response)
+}
+
+// PollWaterConsumption test payload {"name": "dat", "degree": 20.123}
+func PollWaterConsumption(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	payload, err := ioutil.ReadAll(r.Body)
+	defer r.Body.Close()
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusNotAcceptable)
+		return
+	}
+
+	var event devices.WaterConsumptionEvent
+	err = json.Unmarshal(payload, &event)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	waterconsumption := devices.WaterConsumption{}
+	err = waterconsumption.CreateEvent(&event)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	w.WriteHeader(http.StatusCreated)
+
+	fmt.Fprintf(w, "%v", event)
+}
+
+//AllWaterConsumption ...
+func AllWaterConsumption(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	waterconsumptionEvent := devices.WaterConsumption{}
+	device, err := waterconsumptionEvent.GetAll()
+	// testing custom error response
+	if err == devices.ErrNotFound {
+		http.Error(w, errors.New("Events are not found").Error(), http.StatusNotFound)
+		return
+	}
+
+	response, err := json.Marshal(device)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusNotFound)
+		return
+	}
+	w.Header().Set("Content-Type", "application/json")
+	w.Write(response)
+
+}
+
+//QueryWaterConsumption ...
+func QueryWaterConsumption(w http.ResponseWriter, r *http.Request, _ httprouter.Params) {
+	q := r.URL.Query()
+	from := q.Get("from")
```

Comment:
So here you may try to make a time object from that data, try to parse, and if everything is ok - call the DB method